addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -62,10 +74,10 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
             model: "gift.card",
             method: "search_read",
             args: [[["code", "=", this.state.giftCardBarcode]]],
-            fields: ["code", "initial_amount", "balance"],
+            fields: ["code", "initial_amount", "balance","buy_pos_order_line_id"],
           });
           if (giftCard.length) {
-            this.env.pos.giftCard.push(giftCard[0])
+            this.env.pos.giftCard.push(giftCard[0]);
```

Comment:
Don't add a missing semi-colon on a line that is not part of your fix.
It will add unnecessary diff.